### PR TITLE
Add resulsts as object instead of simple string

### PR DESCRIPTION
**Pull request by Toxantron**
The string collection was quite limited and in combination with localized responses also caused failures when confirming instructions in a language different from the time of creation.

### CODE REVIEW

File: src/Moryx.ControlSystem/VisualInstructions/InstructionResult.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Moryx.ControlSystem.VisualInstructions
+{
+    /// <summary>
+    /// Object representing a possible result for an instructions
+    /// </summary>
+    public class InstructionResult
+    {
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/InstructionResult.cs:11]: xml comments for public properties

Key of the result. This value needs to be unique for the instruction the result is used for.

Human readable value of the result

File: src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs [hunk 2]
```diff
@@ -25,6 +25,7 @@ public void GetAllValuesIfNoResultIsDecorated()
 
             // Assert
             Assert.AreEqual(2, instructionResult.Results.Count(), "There should be 2 results because all of the results are not decorated");
+            Assert.AreEqual(2, instructionResult.PossibleResults.Count(), "There should be 2 results because all of the results are not decorated");
```
Comment (1nf0rmagician) [inline src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs:28]: Did you intend to duplicate the line here?

File: src/Moryx.ControlSystem/VisualInstructions/IInstructionResults.cs [hunk 3]
```diff
@@ -34,4 +34,20 @@ public interface IInstructionInputResults : IInstructionResults
...
+    /// </summary>
+    public interface IInstructionKeyResults : IInstructionResults
+    {
+        /// <summary>
+        /// Possibile results for the instructions
+        /// </summary>
+        InstructionResult[] PossibleResults { get; }
+
+        /// <summary>
+        /// Extended <see cref="IInstructionResults.Invoke(string)"/> with filled input object
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/IInstructionResults.cs:49]: I'm not sure whats correct here, but the comment is coppied from the input results . The interface inherits from the IInstruction result however 🤔

File: src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs (hunk 2 above)
Comment (Toxantron) [reply src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs:28]: Kinda, it checks for the new property on the instruction

Comment (1nf0rmagician) [reply src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs:28]: Ahh I missed that detail 👌

File: src/Moryx.ControlSystem/VisualInstructions/IInstructionResults.cs (hunk 3 above)
Comment (Toxantron) [reply src/Moryx.ControlSystem/VisualInstructions/IInstructionResults.cs:49]: thx, extended the comment to indicate the changed result object and purpose

**Pull request merged**